lib/db.js

```diff
@@ -211,8 +220,24 @@ module.exports = (
   }
 
   DB.prototype.sessions = function (uid) {
+    const promises = [
+      this.pool.get('/account/' + uid + '/sessions')
+    ]
+
+    if (this.redis) {
+      promises.push(this.redis.getAsync(uid))
+    }
+
     log.trace({ op: 'DB.sessions', uid: uid })
-    return this.pool.get('/account/' + uid + '/sessions')
+    return P.all(promises)
+      .then(results => {
+        const tokens = results[0]
+        const redisTokens = results[1] ? JSON.parse(results[1]) : []
+        return tokens.map(function (sessionToken) {
+          const redisToken = redisTokens.find(tok => tok.tokenId === sessionToken.tokenId)
+          return redisToken || sessionToken
```

Comment:
sessionTokens contain the following properties that are not in redisToken, however I can add them if we need. Do we need them?

```
{ 
  deviceId: null,
  deviceName: null,
  deviceType: null,
  deviceCreatedAt: null,
  deviceCallbackURL: null,
  deviceCallbackPublicKey: null,
  deviceCallbackAuthKey: null 
}
```